dspace-server-webapp/src/main/java/org/dspace/app/rest/repository/VocabularyEntryDetailsRestRepository.java

```diff
@@ -0,0 +1,94 @@
+/**
+ * The contents of this file are subject to the license and copyright
+ * detailed in the LICENSE and NOTICE files at the root of the source
+ * tree and available online at
+ *
+ * http://www.dspace.org/license/
+ */
+package org.dspace.app.rest.repository;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.apache.commons.lang3.StringUtils;
+import org.dspace.app.rest.Parameter;
+import org.dspace.app.rest.SearchRestMethod;
+import org.dspace.app.rest.exception.LinkNotFoundException;
+import org.dspace.app.rest.exception.RepositoryMethodNotImplementedException;
+import org.dspace.app.rest.model.ResourcePolicyRest;
+import org.dspace.app.rest.model.VocabularyEntryDetailsRest;
+import org.dspace.app.rest.model.VocabularyRest;
+import org.dspace.app.rest.utils.AuthorityUtils;
+import org.dspace.content.authority.Choice;
+import org.dspace.content.authority.ChoiceAuthority;
+import org.dspace.content.authority.Choices;
+import org.dspace.content.authority.service.ChoiceAuthorityService;
+import org.dspace.core.Context;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageImpl;
+import org.springframework.data.domain.Pageable;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.stereotype.Component;
+
+/**
+ * Controller for exposition of vocabularies entry details for the submission
+ *
+ * @author Andrea Bollini (andrea.bollini at 4science.it)
+ */
+@Component(VocabularyRest.CATEGORY + "." + VocabularyEntryDetailsRest.NAME)
+public class VocabularyEntryDetailsRestRepository extends DSpaceRestRepository<VocabularyEntryDetailsRest, String> {
+
+    @Autowired
+    private ChoiceAuthorityService cas;
+
+    @Autowired
+    private AuthorityUtils authorityUtils;
+
+    @PreAuthorize("hasAuthority('AUTHENTICATED')")
+    @Override
+    public Page<VocabularyEntryDetailsRest> findAll(Context context, Pageable pageable) {
+        throw new RepositoryMethodNotImplementedException(ResourcePolicyRest.NAME, "findAll");
+    }
+
+    @PreAuthorize("hasAuthority('AUTHENTICATED')")
+    @Override
+    public VocabularyEntryDetailsRest findOne(Context context, String name) {
```

Comment:
If the name contains a `/`, it seems this is not correctly escaped in the links to this endpoint (e.g. the entry-id in `/api/submission/vocabularyEntryDetails/<:vocabulary-name>:<:entry-id>` contains the `/`)

Reply:
I made some tests, it is not supported right now and it seems complex to manage (I have a branch here with some experiments - see last commits https://github.com/abollini/DSpace/tree/IDwithSlash ).
There are at least two different issues with that:
1. spring boot set by default a filter that prevent / to be urlencoded to enhance security. This can be configured (disabled) and solved.
2. also doing that the spring url mapping is confused by the / as it is decoded before to check the mapping and this will lead to match against our /{rel} subpath (i.e. `/api/submission/vocabularyEntryDetails/vname:entryidwith/something` will search for link repository named something)

as there are not authority implementations so far that have a / in the authority ID I recommend to avoid to deal with this extra complexity until we have a real use case. The solution could be to replace the / in the authority ID in the entry-id with another, less problematic chars like the pipe |

Reply:
The controlled-vocabularies (such as the srsc.xml) can have a slash in their ID. There may be none in default DSpace, but this is intended to be used with local vocabularies.
I understand this is indeed complicated, but it's important to create a ticket for this, and I think it should be solved prior to 7.0 release